### PR TITLE
[ws-manager-bridge] Skip stale prebuild events

### PR DESCRIPTION
**Pull request by easyCZ**
## Description

Do not process stale WorkspaceStatus events when updating prebuilds.

[Metrics](https://grafana.gitpod-staging.com/explore?orgId=1&left=%7B%22datasource%22:%22VictoriaMetrics%22,%22queries%22:%5B%7B%22refId%22:%22A%22,%22instant%22:true,%22range%22:true,%22exemplar%22:false,%22expr%22:%22sum(increase(gitpod_ws_manager_bridge_stale_prebuild_events_total))%22%7D%5D,%22range%22:%7B%22from%22:%221649172321000%22,%22to%22:%22now%22%7D%7D) show that we receive stale events relatively frequently on staging.

Screenshot 2022-04-05 at 21 38 10

This change will result in increased accuracy of our prebuild status.

## Related Issue(s)

Relates to:
* https://github.com/gitpod-io/gitpod/issues/8596

Fixes
* https://github.com/gitpod-io/gitpod/issues/9107

## How to test

Requires a degree of luck, as it depends on event ordering on the message bus, but:
1. Start a prebuild
1. port-forward ws-manager-bridge on port 9500 and fetch metrics with `curl localhost:9500/metrics`
2. Check metrics contain  `gitpod_ws_manager_bridge_stale_prebuild_events_total` with a positive count
1. Check `ws-manager-bridge` logs contain `Stale prebuild event received, skipping.`

## Release Notes

```release-note
ws-manager-bridge skips stale prebuild events
```

/hold

### CODE REVIEW

**Comment by roboquat** [comment]
@easyCZ: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

File: components/ws-manager-bridge/ee/src/bridge.ts [hunk 1]
```diff
@@ -49,31 +53,34 @@ export class WorkspaceManagerBridgeEE extends WorkspaceManagerBridge {
...
             span.setTag("updatePrebuiltWorkspace.prebuildId", prebuild.id);
             span.setTag("updatePrebuiltWorkspace.workspaceInstance.statusVersion", status.statusVersion);
 
             if (prebuild.statusVersion <= status.statusVersion) {
                 // prebuild.statusVersion = 0 is the default value in the DB, these shouldn't be counted as stale in our metrics
                 if (prebuild.statusVersion > 0) {
+                    // We've gotten an event which is younger than one we've already processed. We shouldn't process the stale one.
+                    span.setTag("updatePrebuiltWorkspace.staleEvent", true);
                     this.prometheusExporter.recordStalePrebuildEvent();
+                    return;
```
Comment (easyCZ) [inline components/ws-manager-bridge/ee/src/bridge.ts:71]: This is the only real logic change, rest is formatting from prettier.

**Review by AlexTugarev — APPROVED** [review]
LGTM!

**Comment by easyCZ** [comment]
Thanks for the review @AlexTugarev!

/unhold

**Pull request merged**